examples/truffle/contracts/AbiTypes.sol

```diff
@@ -87,15 +87,15 @@ contract AbiTypes {
     return buf;
   }
 
-  event ValueUint(uint8, uint16, uint32, uint64, uint128, uint256 indexed value);
-  event ValueInt(int8, int16, int32, int64, int128, int256 indexed value);
+  event ValueUint(uint8 a, uint16 b, uint32 c, uint64 d, uint128 e, uint256 indexed value);
```

Comment:
See the commit message for an explanation of why this is needed.

Reply:
Could you point out which stricter size verification is causing this error to happen now?

Reply:
https://github.com/tomusdrw/rust-web3/blob/e6d044a28458be9a3ee31108475d787e0440ce8b/src/contract/tokens.rs#L234

In the web3 version it does not check the value of the integer versus the min/max of the type.

Reply:
I'm not sure I understand why the `int`s don't work. If the value is out of range, its OK:
```rust
    let x = 0xffffffff_ffffffff_ffffffff_ffffffd6u128;
    dbg!(x as i8);
```

And, AFAIK, Solidity ABI ensures these values are sign-extended (and even if they weren't I'm still not sure I understand the problem).

Reply:
In the solidity abi an integer with less 256 bits is still encoded as a u256 or i256. For unsigned integers the value is 0 extended and for signed integers sign extended.

The rust-web3 code does not enforce this as it always takes the low bits without looking at the rest.
My new code does enforce this.

This surfaced a new bug in ethabi where unnamed event variables are not encoded properly https://github.com/rust-ethereum/ethabi/issues/206 .